Packages/com.unity.inputsystem/InputSystem/Devices/Joystick.cs

```diff
@@ -61,7 +61,7 @@ public class Joystick : InputDevice
         /// This is the <see cref="ButtonControl"/> type control on the joystick
         /// that has the <see cref="CommonUsages.PrimaryTrigger"/> usage.
         /// </remarks>
-        public ButtonControl trigger { get; private set; }
+        public ButtonControl trigger { get; protected set; }
```

Comment:
These changes to various devices are required so that the generated C# class can set these properties directly. I could have simply invoked `FinishSetup` like `InputDeviceBuilder` does, but I really wanted to go all the way with this and get rid of all the path lookups done in `FinishSetup` methods through `GetChildControl`. So, `InputControlLayoutGenerator` actually goes and looks at which controls these kind of properties lead to and then just hardwires that directly into the generated class.